packages/fxa-settings/src/lib/channels/firefox.ts

```diff
@@ -380,10 +383,21 @@ export class Firefox extends EventTarget {
           });
         });
       }),
-      new Promise(
-        (resolve) =>
-          (timeout = setTimeout(resolve, DEFAULT_SEND_TIMEOUT_LENGTH_MS))
-      ),
+      // Ideally, we would detect WebChannel support instead of relying on a timeout.
```

Comment:
We should be able to check the config values in metadata and the url possibly to ensure that the webchannel message has a chance to go through. If it doesn't we can fail fast and not wait for a timeout.

Reply:
If there are reliable ways to detect for webchannel support, any reason we didn't do that to begin with?